.github/workflows/ruby.yml

```diff
@@ -0,0 +1,26 @@
+---
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+      - 'KATELLO-*'
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Ruby
+    uses: theforeman/actions/.github/workflows/foreman_plugin.yml@v0
+    with:
+      plugin: katello
+      postgresql_container: ghcr.io/theforeman/postgresql-evr
+      test_existing_database: false
```

Comment:
One thing to consider for branching. When Katello gets branched, you'll have to add a `foreman_version: X.Y-stable` here, as this is what's driving which Foreman branch is used for testing, and you want them to match ;-)

Reply:
Can that be handled somehow centrally through some mapping or matrix file?

Reply:
There is no such thing today (which of course doesn't mean we can't invent it…)

Today, this mapping lives in https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/test/testKatello.groovy#L1 and needs to be edited by whoever does the branching. So it's not a workflow change as such.

Reply:
@wbclark @qcjames53 assuming this merges as-is, we'll need a new branching step to update the `foreman_version` in this `ruby.yml` file.